src/main/scala/kinesis/mock/instances/circe.scala

```diff
@@ -8,14 +8,43 @@ import java.util.Base64
 import io.circe.{Decoder, Encoder}
 
 object circe {
-  implicit val finiteDurationCirceEncoder: Encoder[FiniteDuration] =
-    Encoder[String].contramap(_.toString)
+  val instantLongCirceEncoder: Encoder[Instant] =
+    Encoder[Long].contramap(_.toEpochMilli)
+
+  val instantLongCirceDecoder: Decoder[Instant] =
+    Decoder[Long].map(Instant.ofEpochMilli)
+
+  val instantDoubleCirceEncoder: Encoder[Instant] =
+    Encoder[Double].contramap(x =>
+      java.math.BigDecimal
+        .valueOf(x.toEpochMilli)
+        .scaleByPowerOfTen(-3)
+        .doubleValue()
+    )
+
+  val instantDoubleCirceDecoder: Decoder[Instant] =
+    Decoder[Double].map(x =>
+      Instant.ofEpochMilli(
+        java.math.BigDecimal.valueOf(x).scaleByPowerOfTen(3).longValue()
+      )
+    )
 
-  implicit val instantCirceEncoder: Encoder[Instant] =
-    Encoder[Long].contramap(_.getEpochSecond())
+  val instantBigDecimalCirceEncoder: Encoder[Instant] =
```

Comment:
For some reason, some of the JSON APIs (GetShardIterator, ListShards + filter) use a BigDecimal encoding instead of Double.